src/close_reason.cpp

```diff
@@ -146,8 +138,10 @@ namespace libtorrent
 					return close_no_reason;
 			}
 		}
-		else if (ec.category() == boost::system::system_category())
-		{
+		else if (ec.category() == boost::system::system_category()
+			|| ec.category() == boost::system::generic_category())
+		{	// need to test both categories due to possible boost
```

Comment:
is this caused by boost versions? I would have expected that unix systems would use ``system_category`` and windows ``generic_category``. either way, the intention here was most likely to test for ``generic_category`` to begin with. Actually, would you mind updating this patch to just compare against ``generic_category``? (there's a typedef in the ``libtorrent`` namespace you can use). I should probably look over other places where I compare ``error_codes`` directly.